src/plugins/layout.ts

```diff
@@ -68,6 +68,7 @@ export default class Layout implements IPlugin {
     let discoverTime = getTimestamp();
     traverseNodeTree(document, this.discoverNode.bind(this));
     this.shadowDomConsistent = this.shadowDom.mirrorsRealDom();
+    assert(this.shadowDomConsistent, "discoverDom", "shadowDom inconsistent after dom discovery");
```

Comment:
CAn you return an indication of what inconsistency was? I would encourage you to think about a "diagnostic" approach rather than a simple "pass/fail" approach. You compute diagnostics that indicate issues and can then return what the issues found were and which diagnostics cause a failure. 

This is probably more than the scope of this PR but worth thinking about more.